sql/src/main/java/com/vmware/ddlog/TranslateCreateTableDialect.java

```diff
@@ -22,6 +24,15 @@ protected String visitCreateTable(final CreateTable node, final String sql) {
         for (final TableElement element : node.getElements()) {
             if (element instanceof ColumnDefinition) {
                 final ColumnDefinition cd = (ColumnDefinition) element;
+                String h2Type = cd.getType();
+
+                // We need to strip the array subtype for H2
+                final Pattern arrayType = Pattern.compile("ARRAY\\((.+)\\)");
+                Matcher m = arrayType.matcher(cd.getType());
+                if (m.find()) {
+                    h2Type = "array";
```

Comment:
So there is such a type "array"?
This seems wrong to me.

Reply:
What's your concern? We can discuss before merging.

Neither H2 nor Presto have very well-specified SQL languages, but it appears H2 has a simple "array" type which [accepts no type](http://www.h2database.com/html/datatypes.html#array_type).

Reply:
So perhaps we should add a configuration option for the code generator, indicating the expected SQL back-end.
When we compile for H2 we strip, otherwise we don't.

Reply:
Ah yes, that will be in the next PR. I think I need that for the JooqProvider, because we are doing a silly thing right now where we (manually) convert SQL from H2 to Presto dialect, which gets converted back to H2 inside this ddlog-sql code. Whereas you should just have a flag for whether the current connection is using H2 or Presto, and translate accordingly.

However, I don't necessarily think you need a flag for this particular class. My understanding is that this particular class explicitly assumes it is translating from Presto --> H2.

Reply:
I don't remember any longer, but if that's true, then this is fine.